packages/ComponentLibrary/src/components/PrimaryTab/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react';
+import {
+  Tabs,
+  Tab,
+  Box,
+  Menu,
+  MenuItem,
+  ListItemIcon,
+  Tooltip,
+} from '@mui/material';
+import CheckIcon from '@mui/icons-material/Check';
+import { PrimaryTabsProps, TabItem } from './types';
+import { menuStyle, styles, sx, tabIndicatorProps } from './styles';
+import IconButton from '../IconButton';
+
+const PrimaryTabs: React.FC<PrimaryTabsProps> = ({ tabs, onChange, icon }) => {
+  const [selectedTab, setSelectedTab] = useState(tabs[0]?.id || '');
+  const [hoveredTab, setHoveredTab] = useState<string | null>(null);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const handleChange = (_: React.SyntheticEvent, newValue: string) => {
+    setSelectedTab(newValue);
+    if (onChange) {
```

Comment:
you could improve this check by making sure it's actually a callable object:
```suggestion
    if (onChange && typeof onChange === 'function') {
```